lib/globalize/active_record/instance_methods.rb

```diff
@@ -192,9 +196,9 @@ def save_translations!
       end
 
       def with_given_locale(attributes, &block)
-        attributes.symbolize_keys! if attributes.respond_to?(:symbolize_keys!)
+        symbolized_attributes = attributes.symbolize_keys if attributes.respond_to?(:symbolize_keys)
```

Comment:
I have a small optimisation which ensures that the variable is always set, even if attributes doesn't respond to the method:

```ruby
def with_given_locale(_attrs, &block)
  attributes = _attrs.dup
  attributes.symbolize_keys! if attributes.respond_to?(:symbolize_keys!)

  if locale = attributes.try(:delete, :locale)
    Globalize.with_locale(locale, &block)
  else
    yield
  end
end
```

I stubbed out part of it and tried it in irb and it still works:

```
[29] pry(main)> a
=> {"locale"=>"foo"}
[30] pry(main)> def with_given_locale(_attrs, &block)
[30] pry(main)*   attributes = _attrs.dup
[30] pry(main)*   attributes.symbolize_keys! if attributes.respond_to?(:symbolize_keys!)
[30] pry(main)*   p attributes
[30] pry(main)*   if locale = attributes.try(:delete, :locale)
[30] pry(main)*     "Globalize.with_locale(locale, &block)"
[30] pry(main)*   else
[30] pry(main)*     "yield"
[30] pry(main)*   end
[30] pry(main)* end
=> :with_given_locale
[31] pry(main)> with_given_locale(a)
{:locale=>"foo"}
=> "Globalize.with_locale(locale, &block)"
[32] pry(main)> a
=> {"locale"=>"foo"}
```

Note how it symbolized the keys on the duplicated object inside the method scope but outside it the object still has the same string keys.

@nobuhikosawai What do you think?